pkg/lib/infra/redisqueue/task.go

```diff
@@ -0,0 +1,52 @@
+package redisqueue
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type TaskStatus string
+
+const (
+	TaskStatusPending   TaskStatus = "pending"
+	TaskStatusCompleted TaskStatus = "completed"
+)
+
+type Task struct {
```

Comment:
Should there be a `type` key here to indicate the task type?
(but we have one type of task only for now)

Reply:
Right now it the type is determined by the queue name. A queue only stores one type of task. And each queue has a specific TaskProcessor to process it.